HW02_B_Mimic_starter.Rmd

```diff
@@ -1,99 +1,124 @@
-```
-
-The above graph lets you see some colobrlind friendly palettes. For the graph below, I used Set2.
-
-
-
-Now using the above mpg dataset, make this graph
-
-```{r graph 4 code}
-
-```
-
-
-
-
+---
+title: "HW02_B_Graph-Mimic"
+author: "YOUR NAME HERE"
+output: github_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+```
+
+
+```{r load libraries, echo = TRUE, include=TRUE}
+library("ggplot2")
+library("magrittr") #so I can do some piping
+data("diamonds")
+data("mpg")
+data("iris")
+theme_set(theme_bw()) #I'll give you this one, you can set the theme individually for graphs
+#or you can set the theme for all the graphs using theme_set()
+#theme_bw() is best theme (IMO)
+
+#for graph 3:
+library("ggrepel")
+
+
+```
+
+## HW02 Part B
+
+For this part of the HW, the goal is to try to recreate the graphs I make from scratch. I will only provide the MD, not the actual code I used to create it besides which data I use to create it. The rest will be up to you. 
+
+Try for all 4, but if you are struggling don't worry about it. Try your best for each, if you don't get everything that's what the peer-review is for. :smile:
+
+
+### Graph 1
+```{r, echo = T, include = TRUE}
+data("diamonds")
+#hint think about the *position* the bars are in...
+```
+
+
+Using the diamonds dataset, make this graph:
+```{r graph1 code, echo=FALSE}
+ggplot(diamonds, aes(x = cut, fill = clarity)) +
+  geom_bar(position = "dodge") +
+  labs(title = "My Diamond Collection", x = "Diamond Cut", y = "Number of Diamonds", subtitle = "Boxplot representing the number of diamonds in my diamond collection by \ntype of cut quality and clarity of diamond") +
+  annotate("rect", xmin = 4.5, xmax = 5.5, ymin = 0, ymax = 5000, alpha = 0.3) +
+  annotate("text", x = 3.9, y = 4500, label = "My Best Diamonds, \nof course") +
+  theme(plot.title = element_text(hjust = 0.5))
+
+```
+
+### Graph 2
+```{r, echo = T, include = TRUE}
+data("iris")
+```
+
+Using the iris dataset, make this graph:
+```{r graph 2 code, echo=FALSE}
+ggplot(iris, aes(x = Sepal.Length, y = Petal.Length, shape = Species, color = Species)) +
+  geom_point() +
+  geom_smooth(method = "lm", color = "black", se = FALSE) +
+  facet_wrap(. ~ Species, scales = "free_y") 
+#I do not know how to switch the graphs around in the correct order, if I have time I'll go back and play with this one more later
+```
+
+
+### Graph 3
+You'll need the information in this first box to create the graph
+```{r, echo = TRUE}
+data("mpg")
+corvette <- mpg[mpg$model == "corvette",]
+#install
+require("ggrepel") #useful for making text annotations better, hint hint
+set.seed(42)
+```
+
+Now using the mpg dataset and the corvette dataset, make this graph:
+
+```{r graph 3 code}
+ggplot() +
+  geom_point(data = mpg, mapping = aes(x = displ, y = hwy)) +
+  geom_point(data = corvette, mapping = aes(x = displ, y = hwy), color = "blue") +
```

Comment:
I complicated my life a lot with this one, and your solution is way easier and simple. 

`geom_point(color = ifelse(mpg$model == "corvette", "blue", "black))`

I used the ifelse() function. It saves a line of code, but it took me a long while to find it.